python/python/lance/dataset.py

```diff
@@ -2330,6 +2331,12 @@ def cleanup_old_versions(
             td_to_micros(older_than), delete_unverified, error_if_tagged_old_versions
         )
 
+    def cleanup_with_policy(self, policy: CleanupPolicy) -> CleanupStats:
```

Comment:
Why not just add the new `retain_versions` parameter to `cleanup_old_versions`, why a whole new method?

Reply:
I think it’s fine if the API design differs between Python and Rust. Python has named arguments, which is often a natural API design for Python users. Rust doesn’t, and instead needs to use option structs and builder patterns to make configuration readable.

It seems like this should just be another named argument in Python.